src/bo4e_generator/sqlparser.py

```diff
@@ -0,0 +1,341 @@
+    # clean up imports after removing Field, e.g. from pydantic import Something, \n -> from pydantic import Something\n
+    python_code = python_code.replace(",,", "").replace(", \n", "\n").replace(" , ", " ")
+    python_code = python_code.replace("from pydantic import \n", "")
+    return python_code
+
+
+def make_decimal(field: str) -> str:
+    """
+    turns Union type float or str into Decimal type. For older Schema versions.
+    """
+    return re.sub(r"float \| str", "Decimal", field)
+
+
+def adapt_parse_for_sql(
+    input_directory: Path, namespace: dict[str, SchemaMetadata]
+) -> tuple[dict[str, SchemaMetadata], dict[str, Any], Path, dict[str, str]]:
+    """
+    Scans fields of parsed classes to modify them to meet the SQLModel specifics and to introduce relationships.
+    Returns additional information, an input path with modified json schemas and arguments for the parser
+    """
+    add_relation: DefaultDict[str, dict[str, Any]] = defaultdict(dict)  # added relationship fields
+    relation_imports: DefaultDict[str, dict[str, str]] = defaultdict(dict)  # added imports for relationship fields
+
+    for schema_metadata in namespace.values():
+        if "enum" not in schema_metadata.module_path:
+            # list of fields which will be replaced by modified versions
+            del_fields = []
+            for field, val in schema_metadata.schema_parsed["properties"].items():
+                # modify decimal fields
+                if "number" in str(val) and "string" in str(val):
+                    relation_imports[schema_metadata.class_name + "ADD"]["Decimal"] = "decimal"
+                if "array" in str(val) and "$ref" not in str(val):
+                    add_relation, relation_imports = create_sql_list(
+                        field, schema_metadata.class_name, namespace, add_relation, relation_imports
+                    )
+                    del_fields.append(field)
+                if "$ref" in str(val):  # or "array" in str(val):
+                    add_relation, relation_imports = create_sql_field(
+                        field, schema_metadata.class_name, namespace, add_relation, relation_imports
+                    )
+                    del_fields.append(field)
+            for field in del_fields:
+                del schema_metadata.schema_parsed["properties"][field]
+            # store the reduced version. The modified fields will be added in the BaseModel.jinja2 schema
+            schema_metadata.schema_text = json.dumps(schema_metadata.schema_parsed, indent=2, ensure_ascii=False)
+
+    additional_arguments = additional_sql_arguments(namespace, add_relation, relation_imports)
+    # save intermediate jsons
+    for schema in namespace.values():
+        file_path = (
+            input_directory / Path("intermediate") / Path(schema.module_path[0]) / Path(schema.class_name + ".json")
```

Comment:
```suggestion
            input_directory / "intermediate" / Path(*schema.module_path[:-1]) / schema.module_path[-1] + ".json"
```
Damit das auch gleich beliebige Subpfade unterstützt 😉